fast64_internal/sm64/settings/properties.py

```diff
@@ -0,0 +1,174 @@
+import os
+import bpy
+from bpy.types import PropertyGroup, UILayout, Scene, Context
+from bpy.props import BoolProperty, StringProperty, EnumProperty, IntProperty, FloatProperty, PointerProperty
+from bpy.path import abspath
+from bpy.utils import register_class, unregister_class
+
+from ...render_settings import on_update_render_settings
+from ..sm64_constants import (
+    level_enums,
+    enum_refresh_versions,
+    defaultExtendSegment4,
+    enum_export_type,
+    enum_compression_formats,
+    enum_sm64_goal_type,
+)
+from ..sm64_utility import export_rom_checks, import_rom_checks
+from ...utility import (
+    directory_path_checks,
+    directory_ui_warnings,
+    multilineLabel,
+    prop_split,
+)
+
+
+def decomp_path_update(self, context: Context):
+    directory_path_checks(
+        abspath(self.decomp_path), "Empty decomp path.", "Decomp path does not exist.", "Decomp path is not a folder."
+    )
+    context.scene.fast64.settings.repo_settings_path = os.path.join(
+        os.path.dirname(abspath(self.decomp_path)), "fast64.json"
+    )
+
+
+class SM64_Properties(PropertyGroup):
+    """Global SM64 Scene Properties found under scene.fast64.sm64"""
+
+    version: IntProperty(name="SM64_Properties Version", default=0)
+    cur_version = 2  # version after property migration
+
+    # UI Selection
+    show_importing_menus: BoolProperty(name="Show Importing Menus", default=False)
+    export_type: EnumProperty(items=enum_export_type, name="Export Type", default="C")
+    goal: EnumProperty(items=enum_sm64_goal_type, name="Goal", default="All")
+
+    export_rom: StringProperty(name="Export ROM", subtype="FILE_PATH")
+    output_rom: StringProperty(name="Output ROM", subtype="FILE_PATH")
+
+    extend_bank_4: BoolProperty(
+        name="Extend Bank 4 on Export?",
+        default=True,
+        description=f"\
+Sets bank 4 range to ({hex(defaultExtendSegment4[0])}, {hex(defaultExtendSegment4[1])}) and copies data from old bank",
+    )
+
+    import_rom: StringProperty(name="Import ROM", subtype="FILE_PATH")
+    convertible_addr: StringProperty(name="Address")
+    level_convert: EnumProperty(items=level_enums, name="Level", default="IC")
+
+    decomp_path: StringProperty(name="Decomp Folder", subtype="FILE_PATH", update=decomp_path_update)
+
+    blender_to_sm64_scale: FloatProperty(name="Blender To SM64 Scale", default=100, update=on_update_render_settings)
+
+    # C
+    repo_settings_tab: bpy.props.BoolProperty(default=True)
```

Comment:
it is very confusing that this prop exists both under `SM64_Properties` and `Fast64Settings_Properties`

could you change `SM64_Properties.repo_settings_tab` to `SM64_Properties.sm64repo_settings_tab`

actually is it even intended that you have both? given that `repo_settings_path` and `auto_repo_load_settings` are only in `Fast64Settings_Properties`

Reply:
It still makes it much easier to find things with ctrl F if they are named different

I think the better readability/maintainability is worth the bloat

If you prefer you can find another name that better describes the sm64 setting compared to the general one, something with "saved_settings" maybe?